client/pages/Register.jsx

```diff
@@ -1,203 +1,75 @@
-
+  const [successMessage, setSuccessMessage] = useState(undefined);
   const { control, register, handleSubmit, watch, formState: { errors } } = useForm();
+  const navigate = useNavigate();
 
-  const handleRegistration = (data) => {
-    console.log('REGISTER BUTTON CLICKED', data);
+  onMount(); // Call the onMount function passed as a prop, which sets the subnavbar text
 
-    const apiUrl = 'http://localhost:3000/marvel/registration';
-    let profilePictureUrl = null;
-
-    if (data.file) {
-      console.log('THIS IS THE DATA.FILE', data.file);
-      const formData = new FormData();
-      formData.append('image', data.file);
-      const uploadUrl = 'http://localhost:3000/marvel/upload';
-      axios.post(uploadUrl, formData)
-        .then((response) => {
-          console.log('SUCCESSFUL IMAGE POST', response);
-          profilePictureUrl = `http://localhost:3000/images/${data.file.name}`;
-          console.log('THIS IS THE PATH', profilePictureUrl);
-          data.profilePictureUrl = profilePictureUrl;
-          console.log('BIG DATA', data);
-
-          // Call registration endpoint here
-          axios.post(apiUrl, { ...data, profilePictureUrl })
-            .then((response) => {
-              console.log('SUCCESSFUL POST', response);
-            })
-            .catch((error) => {
-              console.error(error);
-              if (error.response && error.response.status === 409) {
-                const data = error.response;
-                if (data.status === 409) {
-                  setErrorMessage(data.data.error);
-                } else {
-                  setErrorMessage('An error occurred while fetching data. Please try again later.');
-                }
-              } else {
-                setErrorMessage('An error occurred while fetching data. Please try again later.');
-              }
-            });
-        }
-        )
-        .catch((error) => console.error(error));
-    } else {
-      // Call registration endpoint here if no file is present
-      axios.post(apiUrl, data)
-        .then((response) => {
-          console.log('SUCCESSFUL POST', response);
-        })
-        .catch((error) => {
-          console.error(error);
-          if (error.response && error.response.status === 409) {
-            const data = error.response;
-            if (data.status === 409) {
-              setErrorMessage(data.data.error);
-            } else {
-              setErrorMessage('An error occurred while fetching data. Please try again later.');
-            }
-          } else {
-            setErrorMessage('An error occurred while fetching data. Please try again later.');
-          }
-        });
+  const onSubmit = async (data) => {
+    try {
+      await handleRegistration(data, setErrorMessage);
+      setSuccessMessage('Account created successfully. Please wait 5 seconds before navigating to the sign-in page. If you are not redirected, click ');
```

Comment:
Hmm, I wonder what I'm supposed to click! ;-)

Reply:
I have there `here` surrounded by an `<a>` tag with the proper `href` here. It seemed a bit easier to separate the anchor tag to a later section for brevity. Should I adjust it? 

`{successMessage && <h1 className='text-blue-300 bold'>{successMessage} <a href="/sign-in" className='text-blue-500 underline'>here</a></h1>}`